### ISSUE TITLE
Discussion: LLVM implementation of list, set, dict, and tuple

### ISSUE THREAD

**Issue by namannimmo10** [lcompilers/lpython#737]
This is an extension of #488.

We should implement these as early as possible. 

Regarding tuples, Gagandeep and I discussed that we could do those using the concept of structures in the LLVM IR. We have the number of elements and each element type at compile time.

Lists and dictionaries should also be implemented in LLVM directly. 

CC: @certik @Smit-create @czgdp1807

**Comment by czgdp1807** [comment]
Makes sense. Tuples should be easy as they are immutable and all the information is available at compile time.

**Comment by czgdp1807** [comment]
Basically tuples are exactly same as data classes with indexing.

**Comment by namannimmo10** [comment]
This might give a rough idea of what we will be doing: https://mapping-high-level-constructs-to-llvm-ir.readthedocs.io/en/latest/basic-constructs/structures.html

**Comment by namannimmo10** [comment]
I am working on adding tuple support in LLVM.

**Comment by Smit-create** [comment]
I think we can close this. @czgdp1807 Has completed most of the implementation.

**Comment by czgdp1807** [comment]
`set` is left.
